### ISSUE TITLE
Relying on the constructor property to determine the plugin type can easily break

### ISSUE THREAD

**Issue by danielohrlund** [thysultan/stylis.js#125]
The way the type of a plugin is determined in the function `use` https://github.com/thysultan/stylis.js/blob/7c515a7745257cc4ead519b70cba31bfd963726a/stylis.js#L1569-L1584
can easily break as the 'constructor functions'  like Function, Array etc. are not read only. If a program that executes before stylis does something like
```js
Function = function () {
  console.error("You are not allowed to use eval() or new Function()");
};
```
plugins will no longer work as expected.

**Comment by thysultan** [comment]
Have you run into this issue yet(in the wild)? or is it a pre-emptive?

**Comment by danielohrlund** [comment]
I ran into this issue in the wild via https://github.com/styled-components/styled-components.

I work for a security company called tCell and we offer a browser side security tool to our customers that helps them monitor which inline scripts appear in their users browsers (to protect against XSS-attacks). Since both `eval` and `new Function` can be really dangerous if an attacker finds a way to control the inputs to those functions we want to let our users know if their apps are using them, and one of the ways we detect that is by overriding the Function constructor.

**Comment by danielohrlund** [comment]
_Sorry for the double posting but I realized that it is probably better to continue the discussion here since the PR has been merged._

Thanks for accepting the PR! :raised_hands: How often do you issue new releases of the library?

**Comment by thysultan** [comment]
Published in 3.5.4
